libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxUpcomingManeuverAdapter.kt

```diff
@@ -0,0 +1,149 @@
+package com.mapbox.navigation.ui.maneuver.view
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.recyclerview.widget.RecyclerView
+import com.mapbox.navigation.base.formatter.DistanceFormatterOptions
+import com.mapbox.navigation.core.internal.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.ui.base.model.maneuver.Maneuver
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState
+import com.mapbox.navigation.ui.maneuver.R
+import com.mapbox.navigation.ui.maneuver.view.MapboxUpcomingManeuverAdapter.MapboxUpcomingManeuverViewHolder
+import kotlinx.android.synthetic.main.mapbox_layout_main_maneuver_view.view.*
+
+/**
+ * Default recycler adapter to render upcoming maneuvers for the [RouteLeg].
+ * @property context Context
+ * @property inflater (android.view.LayoutInflater..android.view.LayoutInflater?)
+ * @property upcomingManeuverList MutableList<Maneuver>
+ * @constructor
+ */
+class MapboxUpcomingManeuverAdapter(
```

Comment:
Does this need to be `public`? 🤔

Reply:
I don't think so. They already have the data. If the customers are using the view, adapter will be used by default.

Reply:
Should we make it `internal` then?